### PR TITLE
RM-93430 RM-93429 Release over_react 4.1.0

### PR DESCRIPTION
**Pull request by rmconsole-readonly-wk**
Pull Requests included in release:
* [CPLAT-12196: @Component2 annotation parity for function components](https://github.com/Workiva/over_react/pull/655)
* [Consume setup-dart 0.2 action](https://github.com/Workiva/over_react/pull/673)
* [CPLAT-11863 Add Redux Hooks](https://github.com/Workiva/over_react/pull/674)
* [Dart SDK 2.9+ boilerplate compatibility](https://github.com/Workiva/over_react/pull/679)

Requested by: @joebingham-wk

Diff Between Last Tag and Proposed Release: https://github.com/Workiva/over_react/compare/4.0.0...Workiva:release_over_react_4.1.0
Diff Between Last Tag and New Tag: https://github.com/Workiva/over_react/compare/4.0.0...4.1.0

The logs for the request that created this PR can be found [here](https://w-rmconsole.appspot.com/release/release_event/6449952306233344/logs/)
This pull request can be recreated by clicking [here](https://w-rmconsole.appspot.com/api/v2/rosie/reTriggerReleaseEvents/6449952306233344/?pull_number=680&repo_name=Workiva%2Fover_react)

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
 
+## [4.1.0](https://github.com/Workiva/over_react/compare/4.0.0...4.1.0)
+
+- [#679] Update boilerplate to be compatible with Dart >=2.9.0.
+
+  This introduced a new component factory syntax that future proofs the boilerplate syntax through Dart 2.12! For more information and migration instructions, refer to the [PR description](https://github.com/Workiva/over_react/pull/679).
+
+- [#674] Add Redux hooks. 
+
+  Introducing new Redux hook APIs! OverReact consumers also using OverReact Redux can now use the `useDispatch`, `useSelector`, and `useStore` hooks. For more information, refer to the [documenation](https://github.com/Workiva/over_react/blob/3e968e80bcea7a62a179a13930765b126c767254/doc/over_react_redux_documentation.md#hooks).
```
Comment (aaronlademann-wf) [inline CHANGELOG.md:11]:
```suggestion
  Introducing new Redux hook APIs! OverReact consumers also using OverReact Redux can now use the `useDispatch`, `useSelector`, and `useStore` hooks. For more information, refer to the [documenation](https://github.com/Workiva/over_react/blob/master/doc/over_react_redux_documentation.md#hooks).
```

**Review by aaronlademann-wf — CHANGES_REQUESTED** [review]

**Review by aaronlademann-wf — APPROVED** [review]
+1

**Comment by joebingham-wk** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**